libraries/AP_GyroFFT/AP_GyroFFT.cpp

```diff
@@ -540,10 +544,21 @@ void AP_GyroFFT::update_parameters(bool force)
     }
 #endif
 
+    if (_fft_min_hz >= _fft_max_hz) {
+        // the parameters were just configured badly
```

Comment:
You will need to set valid to false here, you also need to do it inside the semaphore

Reply:
At the moment we'll get a pre-arm failure in the case of failure - I thought it would generally be a Bad Idea to fly as the filter might be critical for the aircraft's stability...

Note there's no way back from that - if you fail configuration you get to fix it before arming.

I mean, I can make this boolean, but I'd probably just be assigning into `_config.valid` in the caller anyway, and that spreads the knowledge of what `_config` looks like out a bit.

Happy to make it boolean and assign into `_config.valid` anyway - do you want me to do that?

Reply:
You don't need config valid if you do what I put. If we are going to change the meaning and logic of this function it needs to be a separate PR - this is a backport so lets do the minimal